ocaml/runtime/dynlink_nat.c

```diff
@@ -51,6 +51,17 @@ static void *getsym(void *handle, const char *module, const char *name){
   return sym;
 }
 
+static void *getsym_exn(void *handle, const char *module, const char *name){
+  void *sym = getsym(handle, module, name);
+  if (sym == NULL) {
+    value msg = caml_alloc_sprintf(
+      "Dynlink: failed to load module due to missing symbol %s%s",
+      module, name);
+    caml_invalid_argument_value(msg);
```

Comment:
What happens after this exception has raised?  We might have partially-registered shared libraries depending on which of the calls below we're in.  Maybe this is ok because the duplicate loading protection is via `caml_register_dyn_global` I think.

Reply:
Those libraries will be in a bad state but it won't affect anything else. (Libs aren't unloaded, even on failure to load, so if e.g. useless frametable entries are added it's fine)